spring-cloud-starter-stream-processor-tasklaunchrequest-transform/src/main/java/org/springframework/cloud/stream/app/tasklaunchrequest/transform/processor/TasklaunchrequestTransformProcessorConfiguration.java

```diff
@@ -85,15 +88,22 @@ public Object setupRequest(String message) {
 		if(StringUtils.hasText(processorProperties.getApplicationName())) {
 			applicationName = processorProperties.getApplicationName();
 		}
+		if(StringUtils.hasText(processorProperties.getEnvironmentProperties())) {
+			properties = parse(processorProperties.getEnvironmentProperties());
+		}
 
 		TaskLaunchRequest request = new TaskLaunchRequest(
 				processorProperties.getUri(),
 				commandLineArgs,
 				properties,
 				deploymentProperties,
 				applicationName);
-
-		return request;
+		Map<String, Object> headerMap = new HashMap<>(message.getHeaders());
+		headerMap.remove("contentType");
```

Comment:
Ok, for the idiot in the room (me), why is this required?  It feels *very* counter-intuitive.  Also, stupid question...is there a constant we should be using here to refer to the header `contentType`?  This feels like something standard enough to have centralized...

Reply:
Oh! Great! I'm not alone in the boat 😄 

Right, there is `MessageHeaders.CONTENT_TYPE` for this purpose.

Reply:
So, the reason for it is this:
- Header-set content-type takes precedence over any other content-type directive, so we either hav to set content type here or rely on the default which is `application/json` and the one and only one required here.
Doe that help?